### DISCUSSION TITLE
Articoli assenti nella fattura elettronica creata in Api 2

### DISCUSSION THREAD

**Discussion by EdoFir** [fattureincloud/api#147] [Q&A] [answered]
Ciao. Sto migrando un client che esporta fatture elettroniche in Fatture in Cloud. Mi succede che la fattura elettronica mi viene creata, senza errori, ma senza articoli (e con importo a zero). Ma il json che mando sembra avere tutti i dati necessari (che riporto di seguito). Qualcuno ha idea di quale può essere  la causa o come cercare di risolverlo? Grazie in anticipo. Edo
{
	"data": {
		"type": "invoice",
		"entity": {
			"name": "XXXXXXXXXXXXXXXXXXXXXX",
			"vat_number": "XXXXXXXXXXXX",
			"tax_Code": "XXXXXXXXXX",
			"address_street": "XXXXXXXXXXXXXXXXX",
			"address_postal_code": "50052",
			"address_city": "CERTALDO",
			"address_province": "FI",
			"country": "Italia",
			"ei_code": "XXXXXXX"
		},
		"date": "2022-07-05",
		"number": "15",
		"numeration": "/G04122999",
		"currency": {
			"id": "EUR",
			"exchange_rate": "1.00000",
			"symbol": "€"
		},
		"language": {
			"code": "it",
			"name": "italiano"
		},
		"use_gross_prices": true,
		"e_invoice": true,
		"ei_data": {
			"payment_method": "MP01"
		},
		"item_list": [
			{
				"name": ".SPRITE LATTINA",
				"description": "SCONTRINO del 05/07/2022 - 99IEC009324 0019-0047",
				"measure": "ML",
				"code": "0437996",
				"qty": "18.00000000",
				"net_price": "0.451000",
				"gross_price": "0.550145",
				"vat": {
					"id": 0
				}
			},
			{
				"name": "ALLUMINIO 150 METRI REAL CASA",
				"description": "SCONTRINO del 05/07/2022 - 99IEC009324 0019-0047",
				"measure": "PZ",
				"code": "0021546",
				"qty": "2.00000000",
				"net_price": "9.754120",
				"gross_price": "11.900026",
				"vat": {
					"id": 0
				}
			},
			{
				"name": "BANANE SFUSE",
				"description": "SCONTRINO del 05/07/2022 - 99IEC009324 0019-0047",
				"measure": "KG",
				"code": "0860080",
				"qty": "1.25800000",
				"net_price": "1.230800",
				"gross_price": "1.280032",
				"vat": {
					"id": 4
				}
			},
			{
				"name": "CAVOLI CAPPUCCIO",
				"description": "SCONTRINO del 05/07/2022 - 99IEC009324 0019-0047",
				"measure": "KG",
				"code": "0860530",
				"qty": "5.71600000",
				"net_price": "0.951960",
				"gross_price": "0.990038",
				"vat": {
					"id": 4
				}
			},
			{
				"name": "INSALATA GENTILE",
				"description": "SCONTRINO del 05/07/2022 - 99IEC009324 0019-0047",
				"measure": "KG",
				"code": "0860628",
				"qty": "3.78000000",
				"net_price": "2.548080",
				"gross_price": "2.650003",
				"vat": {
					"id": 4
				}
			},
			{
				"name": "LATTE INTERO 100% ITALIANO",
				"description": "SCONTRINO del 05/07/2022 - 99IEC009324 0019-0047",
				"measure": "LT",
				"code": "0437549",
				"qty": "2.00000000",
				"net_price": "1.144240",
				"gross_price": "1.190010",
				"vat": {
					"id": 4
				}
			},
			{
				"name": "LATTE INTERO UHT BONTAL",
				"description": "SCONTRINO del 05/07/2022 - 99IEC009324 0019-0047",
				"measure": "LT",
				"code": "0069921",
				"qty": "12.00000000",
				"net_price": "0.817320",
				"gross_price": "0.850013",
				"vat": {
					"id": 4
				}
			},
			{
				"name": "MELONE ITALIANO",
				"description": "SCONTRINO del 05/07/2022 - 99IEC009324 0019-0047",
				"measure": "KG",
				"code": "0860252",
				"qty": "1.92800000",
				"net_price": "1.240400",
				"gross_price": "1.290016",
				"vat": {
					"id": 4
				}
			},
			{
				"name": "NETTARE DI GUAVA PFANNER",
				"description": "SCONTRINO del 05/07/2022 - 99IEC009324 0019-0047",
				"measure": "LT",
				"code": "0432508",
				"qty": "2.00000000",
				"net_price": "1.221380",
				"gross_price": "1.490084",
				"vat": {
					"id": 0
				}
			},
			{
				"name": "POMODORO GRAPPOLO",
				"description": "SCONTRINO del 05/07/2022 - 99IEC009324 0019-0047",
				"measure": "KG",
				"code": "0860738",
				"qty": "10.51800000",
				"net_price": "0.951960",
				"gross_price": "0.990038",
				"vat": {
					"id": 4
				}
			},
			{
				"name": "SHOPPER ORTOFRUTTA 60FR",
				"description": "SCONTRINO del 05/07/2022 - 99IEC009324 0019-0047",
				"measure": "PZ",
				"code": "0444459",
				"qty": "5.00000000",
				"net_price": "0.008240",
				"gross_price": "0.010053",
				"vat": {
					"id": 0
				}
			}
		],
		"payment_list": [
			{
				"amount": 79.5,
				"due_date": "2022-07-05",
				"paid_date": "2022-07-05"
			}
		]
	}
}

**Comment by emajo** [comment, marked as answer]
Ciao @EdoFir, dal json che hai inviato i campi `items_list` e `payments_list` sono chiamati invece `item_list` e `payment_list`.

Facci sapere se riesci a risolvere, 
Grazie,
Emanuele.

**Reply by EdoFir** [reply]
Gentilissimo Emanuele,
grazie! Ho risolto grazie alla tua segnalazione. Mi ha un po' sorpreso che non sia stato emesso un errore. In ogni caso.... Mi sembra di capire che un buon modo di fare il debug può essere quello di confrontare il json che invio io con quello che viene restituito nella responseText con tutti i valori di default che vengono attribuiti. Giusto?
Edo

**Reply by emajo** [reply]
Ciao @EdoFir, i campi diversi da quelli accettati vengono semplicemente ignorati e, siccome items_list e payments_list possono essere vuoti non viene restituito nessun errore.
Per quanto riguarda il debug delle richieste si, quello di confrontare il json inviato con quello ricevuto è un buon modo, ma per velocizzare ancora di più lo sviluppo ti consiglio, per esempio per la creazione di una fattura, di crearla da interfaccia web e recuperarla facendo una chiamata get del documento creato per avere degli esempi di come creare i documenti senza dover capire a cosa servono tutti i campi.
